### DISCUSSION TITLE
Make the cloud apps installable via composer for automated deployment

### DISCUSSION THREAD

**Discussion by GautierArc** [shopware/platform#3158] [RFC / Proposal]
### Description: 
Make the cloud apps installable via composer for automated deployment.

### Benefits
Currently, the cloud apps installation cannot be integrated in an automated deployment process.

**Comment by OliverSkroblin** [comment]
Hi @GautierDig,
thanks for the proposal. Could you perhaps go into a bit more detail? Otherwise there is a lot of room for interpretation regarding this proposal. This could lead to an implementation that looks completely different from the one you have in mind.

**Comment by GautierArc** [comment]
Hi @OliverSkroblin,
Basically, the cloud apps installation and activation should be working exactly the same as the non cloud apps to make it easy to integrate the installation of cloud apps in an automated deployment script.

**Comment by shyim** [comment]
You could use `store:download` to download that app in the CI. Having apps in composer increases the version fragmentation, what we definitely don't want as backend server needs probably to support them.

**Comment by GautierArc** [comment]
Is there a way to specify the version with the `store:download` command ?

**Comment by lasomethingsomething** [comment]
Thank you for bringing this idea to us, @GautierDig. Closing this discussion now due to lack of activity in recent years.

**Reply by mitelg** [reply]
it is possible to install apps via composer since a while

**Reply by JoshuaBehrens** [reply]
It is still not possible to license them that way. That is difficult to handle regarding store credentials in CI/CD

**Reply by shyim** [reply]
When you do this https://developer.shopware.com/docs/guides/hosting/installation-updates/deployments/deployment-helper.html#automatic-store-login

the username / password will be replaced this year with an token, when we fully migrated to ory
